src/include/btree_inline.h

```diff
@@ -64,6 +64,23 @@ __wt_evict_page_soon_check(WT_SESSION_IMPL *session, WT_REF *ref, bool *inmem_sp
     btree = S2BT(session);
     page = ref->page;
 
+    if (!__wt_evict_page_is_soon_or_wont_need(page))
+        return (false);
+
+    /*
+     * Don't schedule pages that will require an update-restore eviction. Wait for them to be old
+     * enough to get evicted.
+     */
+    if (!F_ISSET(S2C(session)->evict, WT_EVICT_CACHE_DIRTY_HARD | WT_EVICT_CACHE_UPDATES_HARD) &&
+      F_ISSET_ATOMIC_64(S2C(session), WT_CONN_PRECISE_CHECKPOINT) &&
+      !__wt_evict_aggressive(session) && __wt_page_is_modified(page)) {
```

Comment:
I think the aggressive check becomes redundant when the `_HARD` checks are in place. It can be removed.

I also suspect that this change is still stronger than necessary. Could you add a statistic - something like "Urgent eviction skipped due to recent updates on page"